finetune/target_models/deployment_model.py

```diff
@@ -0,0 +1,340 @@
+        self.validate_config()
+        self.input_pipeline = DeploymentPipeline(self.config)
+        super().__init__(**kwargs)
+        self.config.base_model = base_model
+        self.task = TaskMode.CLASSIFICATION
+        self.input_pipeline.task = self.task
+        self.featurizer_loaded = False
+        self.adapters = False
+
+    def load_featurizer(self):
+        """
+        Performs graph compilation of the featurizer, saving most compilation overhead from occurring at predict time. Should
+        be called after initialization but BEFORE any calls to load_trainables or predict.
+        """
+        self.featurizer_est = self._get_estimator('featurizer')
+        self.predict_hooks.feat_hook.model_portion = 'whole_featurizer'
+        for hook in self.predict_hooks:
+            hook.need_to_refresh = True
+        output = self.predict(['finetune'], exclude_target=True) #run arbitrary predict call to compile featurizer graph
+        self.featurizer_loaded = True
+
+    def load_trainables(self, path):
+        """
+        Load in a model from save file. Must be called after 
+        """
+        if not self.featurizer_loaded:
+            raise FinetuneError('Need to call load_featurizer before loading weights from file.')
+        original_model = self.saver.load(path)
+        if original_model.config.adapter_size is None:
+            LOGGER.warning("Loading without adapters will result in slightly slower prediction than models that use adapters.")
+            self.predict_hooks.feat_hook.model_portion = 'whole_featurizer' #need to load everything from save file, rather than standard base model file
+        self._target_model = original_model._target_model
+        self._predict_op = original_model._predict_op
+        self._predict_proba_op = original_model._predict_proba_op
+        for hook in self.predict_hooks:
+            hook.need_to_refresh = True
+        self._to_pull = 0
+
+        self.input_pipeline.target_dim = original_model.input_pipeline.target_dim
+        self.input_pipeline.label_encoder = original_model.input_pipeline.label_encoder
+        self.input_pipeline.text_encoder = original_model.input_pipeline.text_encoder
+        self.input_pipeline._target_encoder = original_model.input_pipeline._target_encoder
+        self.input_pipeline._post_data_initialization = original_model.input_pipeline._post_data_initialization
+        self.input_pipeline._format_for_inference = original_model.input_pipeline._format_for_inference
+        self.input_pipeline._format_for_encoding = original_model.input_pipeline._format_for_encoding
+
+        self.adapters = original_model.config.adapter_size is not None
+        
+        if isinstance(original_model.input_pipeline, SequencePipeline) :
+            self.task = TaskMode.SEQUENCE_LABELING
+            self.input_pipeline.multi_label = original_model.input_pipeline.multi_label
+            self.multi_label = original_model.config.multi_label_sequences
```

Comment:
Is there a good reason we don't use `.config.multi_label_sequences` here rather than creating a new attribute on `self`?

Reply:
When we call `SequenceLabeler.predict(self, x)`, the method within SequenceLabeler requires a self.multilabel attribute. The other option is to modify SequenceLabeler, or duplicate its logic, but this seemed cleaner.